src/schemas/tokens-info.ts

```diff
@@ -26,34 +26,61 @@ const checkMinVersion: CustomValidator = (value) => {
   }
 }
 
-const TokensInfoSchema = Joi.object().pattern(
-  AddressSchema,
+const BaseTokenInfoSchema = Joi.object({
+  address: AddressSchema,
+  imageUrl: Joi.string()
+    // For now only allow assets within this repo
+    .pattern(
+      /^https:\/\/raw.githubusercontent.com\/valora-inc\/address-metadata\/main\/assets\/tokens\/[^/]+\.png$/,
+    )
+    .uri()
+    .custom(checkMatchingAsset, 'has a matching asset'),
+  name: Joi.string().required(),
+  decimals: Joi.number().required(),
+  symbol: Joi.string().required(),
+  isCoreToken: Joi.boolean(),
+  isSupercharged: Joi.boolean(),
+  pegTo: AddressSchema,
+  // Forcing flag to be false if decimals is not 18
+  isSwappable: Joi.when('decimals', {
+    is: 18,
+    then: Joi.boolean(),
+    otherwise: Joi.valid(false),
+  }),
+  minimumAppVersionToSwap: Joi.string()
+    .pattern(/^\d+\.\d+\.\d+$/)
+    .custom(checkMinVersion, 'has a valid version'),
+  isNative: Joi.boolean(),
+})
+
+export const TokenInfoSchema = Joi.alternatives().try(
+  Joi.object({
+    // native tokens don't have an address except CELO
+    isNative: Joi.valid(true).required(),
+    symbol: Joi.string().invalid('CELO').required(),
+    address: Joi.forbidden(),
+  }).concat(BaseTokenInfoSchema),
   Joi.object({
+    // CELO is native and has an address
+    isNative: Joi.valid(true).required(),
+    symbol: Joi.valid('CELO').required(),
     address: AddressSchema.required(),
-    imageUrl: Joi.string()
-      // For now only allow assets within this repo
-      .pattern(
-        /^https:\/\/raw.githubusercontent.com\/valora-inc\/address-metadata\/main\/assets\/tokens\/[^/]+\.png$/,
-      )
-      .uri()
-      .custom(checkMatchingAsset, 'has a matching asset'),
-    name: Joi.string().required(),
-    decimals: Joi.number().required(),
-    symbol: Joi.string().required(),
-    isCoreToken: Joi.boolean(),
-    isSupercharged: Joi.boolean(),
-    // This checks it's referencing a token address that exists in the root object
-    pegTo: Joi.valid(Joi.ref('/', { in: true })),
-    // Forcing flag to be false if decimals is not 18
-    isSwappable: Joi.when('decimals', {
-      is: 18,
-      then: Joi.boolean(),
-      otherwise: Joi.valid(false),
-    }),
-    minimumAppVersionToSwap: Joi.string()
-      .pattern(/^\d+\.\d+\.\d+$/)
-      .custom(checkMinVersion, 'has a valid version'),
-  }),
+  }).concat(BaseTokenInfoSchema),
+  Joi.object({
+    // all non-native tokens require address
+    isNative: Joi.boolean().invalid(true),
+    symbol: Joi.string().invalid('CELO').required(),
+    address: AddressSchema.required(),
+  }).concat(BaseTokenInfoSchema),
 )
 
-export default TokensInfoSchema
+export const RTDBAddressToTokenInfoSchema = Joi.object().pattern(
+  AddressSchema,
+  BaseTokenInfoSchema.concat(
+    Joi.object({
+      address: AddressSchema.required(),
+      // This checks it's referencing a token address that exists in the root object
+      pegTo: Joi.valid(Joi.ref('/', { in: true })),
```

Comment:
Why do we need this special case for `pegTo` just here?

Reply:
This is checking that the pegTo address is a key of one of the other tokenInfo values in the RTDB object mapping addresses to tokenInfo objects. 

The BaseTokenInfoSchema is just for tokenInfo objects. Currently it's just used to validate an array of tokenInfo objects one by one. 

To get the pegTo address validation for the general (non-RTDB) case, we could:
1. update our API to make it easier for Joi to validate
2. write our own validation logic with Jest for this

I think the API is cleaner as-is so I prefer 2, though it's a little annoying we can't get all the validation we might want just by using Joi that way

Reply:
I added a jest test for pegTo. It's kinda ugly, but I think it will do the job..

Reply:
Probably we should migrate this `pegTo` to use the tokenId convention? So we can point to native tokens?
Also, we could potentially peg the value to a token in another chain?

Reply:
Until there's a use case for it, I'm inclined to keep pegTo as-is. It's [only used by blockchain-api](https://github.com/search?q=org%3Avalora-inc%20pegto&type=code), so we could make breaking changes to it if we needed to without having to worry as much about backwards compatibility.

Reply:
ok actually, from looking at how blockchain-api consumes this data, it looks like the address-metadata cloud function should use token ID's for pegTo, as you suggested... otherwise we'd have trouble organizing our prices information, which is currently a map from addresses to price strings (ugh!)

Reply:
I think being on Celo and having every token have its own address has made us soft

Reply:
I'd like to deal with pegTo in a separate PR, since I think that will take its own thinking, and can be done pretty independently from the main thrust of this PR. sound good @dievazqu ?